### PR TITLE
Smooth movement - "TG coders suck" edition

### PR DESCRIPTION
**Pull request by monster860**
adds smooth movement! TG tried to do this but then they reverted it. Why? Because their code sucked!

- mob's glide size was based off the completely wrong number (cached_multiplicative_slowdown was not the correct number, and this caused many problems)
- all glide_size was multiplied by 1.25 for some reason
- ghosts had wrong glide_size
- no LONG_GLIDE so diagonal movement was poo-poo
- transit tubes, space-move, and disposals weren't properly coded

in short, the TG coder that originally wrote this was clearly licking chalkboard erasers

[Video!](https://cdn.discordapp.com/attachments/205784753352343552/697626244795334657/out.mp4)

:cl: monster860
add: Movement is now smooth
/:cl:

### CODE REVIEW

**Comment by Marmio64** [comment]
you just hate that coder with a passion dont you?

**Comment by Bibby0110** [comment]
I think you need a video of this in action.

**Comment by Codeatmos** [comment]
This is a pretty big change,I would really like to see a video of it.

**Comment by monster860** [comment]
https://cdn.discordapp.com/attachments/205784753352343552/697626244795334657/out.mp4

**Comment by athena413** [comment]
yes

**Comment by Iamgoofball** [comment]
im gonna steal this once tg's repo is back up, thanks
